cabal-install/src/Distribution/Client/ProjectBuilding/Types.hs

```diff
@@ -192,6 +192,7 @@ instance Exception BuildFailure
 -- | Detail on the reason that a package failed to build.
 --
 data BuildFailureReason = DependentFailed PackageId
+                        | GracefulFailure String
```

Comment:
Same as above, BuildFailureReason is exported. https://hackage.haskell.org/package/cabal-install-3.8.1.0/docs/Distribution-Client-ProjectBuilding-Types.html#t:BuildFailureReason